sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveTableScanSuite.scala

```diff
@@ -192,4 +192,47 @@ class HiveTableScanSuite extends HiveComparisonTest with SQLTestUtils with TestH
       case p: HiveTableScanExec => p
     }.get
   }
-}
+
+  test("Test the InputFormat optimizer") {
+    withTable("table_old", "table_pt_old", "table_new", "table_pt_new") {
+      sql("set spark.sql.hive.fileInputFormat.enabled=true")
+      sql("set spark.sql.hive.fileInputFormat.split.maxsize=134217728")
+      sql("set spark.sql.hive.fileInputFormat.split.minsize=134217728")
+      sql(
+        s"""
+           |CREATE TABLE table_old (id int)
+           |STORED AS
+           |INPUTFORMAT 'org.apache.hadoop.mapred.TextInputFormat'
+           |OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'
+         """.stripMargin)
+      sql(
+        s"""
+           |CREATE TABLE table_pt_old (id int)
+           |PARTITIONED BY (a int, b int)
+           |STORED AS
+           |INPUTFORMAT 'org.apache.hadoop.mapred.TextInputFormat'
```

Comment:
Aren't you able to use the table format to combined table format in Hive table instead?

Reply:
yeah, we can set this when we create a table. However, in some cases, we need to decide whether using combined format or not for some reasons(eg. the size of splits..). And as for those tables we have created, we don't need to modify the table structure.

Reply:
If it's already possible, let's don't do this. That's the downside of using `CombineTextInputFormat` at the first place? We can control the split size via configurations.

Reply:
I think you are right, it seems unnecessary to do that.. Next time I'll think more before start coding, thank you for your time!

Reply:
That's fine. Thanks for your contribution.